File: VB/GroupSorting/RandomStringHelper.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text
Imports System.Threading.Tasks

Namespace DXSample
	Public Class RandomStringHelper
		Private Shared rnd As New Random()
		Private Shared letters As String = "abcdefghijklmnopqrstuvwxyz"

		Public Shared Function GetRandomString() As String
			Dim length As Integer = rnd.Next(6, 20)
			Dim retVal As String = ("" & letters.Chars(rnd.Next(25))).ToUpper()

			For i As Integer = 0 To length - 2
				retVal &= letters.Chars(rnd.Next(25))
			Next i

			Return retVal
		End Function
	End Class
End Namespace

File: CS/GroupSorting/RandomStringHelper.cs
// Developer Express Code Central Example:
// How to apply sorting by Summary along with sorting by the Column.
// 
// This example shows how to sort group summary items along with the column that
// the summary is aligned with.
// To add this functionality, you can use the
// proposed helper class (GroupSummaryOptions) in your solution. Just set the
// attached property of the Grid to "True".
// 
// You can find sample updates and versions for different programming languages here:
// http://www.devexpress.com/example=E4611

using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace DXSample {
    public class RandomStringHelper {
        static Random rnd = new Random();
        static string letters = "abcdefghijklmnopqrstuvwxyz";

        public static string GetRandomString(){
            int length = rnd.Next(6, 20);
            string retVal = ("" + letters[rnd.Next(25)]).ToUpper();

            for (int i = 0; i < length - 1; i++)
                retVal += letters[rnd.Next(25)];

            return retVal;
        }
    }
}
